docs/standard/native-interop/best-practices.md

```diff
@@ -251,6 +251,59 @@ finally
 }
 ```
 
+## Windows and non-Windows data type considerations
+
+There are types in the C language that have latitude in how they are defined. When writing cross-platform interop, cases can arise where platforms differ and can cause issues if not considered.
+
+### C `long`
+
+The `long` in C is defined to have ["at least 32"](https://en.cppreference.com/w/c/language/arithmetic_types) bits. This means there is a minimum required but platforms are not limited to that. The table below illustrates the differences in provided bits for the C `long` data type between platforms.
+
+|             | 32-bit | 64-bit |
+|:------------|:-------|:-------|
+| Windows     | 32     | 32     |
+| macOS/\*nix | 32     | 64     |
+
+These differences make authoring cross-platform P/Invokes difficult when the native function is defined to use `long` on all platforms. It is recommended for .NET 5 and prior .NET version to declare a Windows and a non-Windows signature to handle the problem.
+
+```csharp
+    // Cross platform C function
+    // void Function(long a);
+#if WINDOWS
+    [DllImport("NativeLib", EntryPoint = "Function")]
+    extern static void Function32(int a);
+    [DllImport("NativeLib", EntryPoint = "Function")]
+    extern static void Function64(int a);
+#else
+    [DllImport("NativeLib", EntryPoint = "Function")]
+    extern static void Function32(int a);
+    [DllImport("NativeLib", EntryPoint = "Function")]
+    extern static void Function64(long a);
+#endif
+    
+// Usage
+if (IntPtr.Size == 4)
+{
+    Function32(a);
+}
+else
+{
+    Function64(a);
+}
+```
+
+In .NET 6 a new type has been introduced to help ameliorate this issue - [`CLong` and `CULong`](https://github.com/dotnet/runtime/issues/13788). In .NET 6 the following is now possible.
```

Comment:
Ideally we'd have a doc to link to. @carlossanlop there are XML comments in the files [eg](https://github.com/dotnet/runtime/blob/1ee59da9f6104c611b137c9d14add04becefdf14/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/CLong.cs#L17-L25) should I expect that there's a page on docs.microsoft.com that will appear for these?

Reply:
This public type was [introduced on Jan 12 2021](https://github.com/dotnet/runtime/pull/46401). We have not yet created any ".NET 6.0 Preview" entry in the [Version dropdown of MS Docs](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices?view=net-5.0).

@AaronRobinsonMSFT I suggest you create an issue and add it to your backlog, so you can update this document and have the link point to the .NET 6.0 MS Docs API page of the `CLong` type, after we create the first Preview dropdown item.